### PR TITLE
Host tensor 2 vector refactor

### PR DESCRIPTION
**Pull request by pszmel**
Size of the libngraph.so did not increase after refactoring. The main goal of refactoring was to increase readability.

### CODE REVIEW

File: ngraph/core/src/op/variadic_split.cpp [hunk 1]
```diff
@@ -171,57 +171,24 @@ namespace
...
-        }
-        case element::Type_t::u64:
-        {
-            const auto split_lengths_u64 = read_vector<uint64_t>(split_lengths_tensor);
-            split_lengths =
-                std::vector<int64_t>(std::begin(split_lengths_u64), std::end(split_lengths_u64));
-            break;
-        }
-        default:
-            NODE_VALIDATION_CHECK(split_node,
```
Comment (jdanieck) [inline ngraph/core/src/op/variadic_split.cpp:218]: Missing split_lenghts_tensor type validation .

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by jdanieck** [comment]
Looks like you included unnecessary commits into your branch (5d14f8e, f5d7647,  8d27af8). If this is mistake please clean it up.

**Review by jdanieck — APPROVED** [review]

**Review by blesniewski — APPROVED** [review]

**Comment by blesniewski** [comment]
I'm not sure I like doing error handling like this in switch statements, but it's not wrong either.

**Comment by pszmel** [comment]
> I'm not sure I like doing error handling like this in switch statements, but it's not wrong either.

We talked about it with @jdanieck and thought about adding a helper function but I thought that it wouldn't make it much cleaner, so I decided to go with this idea

File: ngraph/core/src/op/split.cpp [hunk 2]
```diff
@@ -222,14 +222,11 @@ namespace
                         const int64_t num_splits,
                         const Node* split_node)
     {
-        int64_t axis;
         switch (axis_tensor->get_element_type())
```
Comment (mbencer) [inline ngraph/core/src/op/split.cpp:225]: According to spec, axis can be any integer number (only i32, i64, u64 was used to reduce binary size) so please replace it with `is_integral_number()` assert.

File: ngraph/core/src/op/variadic_split.cpp [hunk 3]
```diff
@@ -171,15 +171,12 @@ namespace
                                  const HostTensorVector& outputs,
                                  const Node* split_node)
     {
-        int64_t axis;
         switch (axis_tensor->get_element_type())
```
Comment (mbencer) [inline ngraph/core/src/op/variadic_split.cpp:174]: as above

File: ngraph/core/src/op/variadic_split.cpp [hunk 4]
```diff
@@ -188,32 +185,16 @@ namespace
                                   " during evaluate Split:v1");
             break;
         }
+
+        int64_t axis = host_tensor_2_vector<int64_t>(axis_tensor)[0];
+
         axis = ngraph::normalize_axis(split_node, axis, data_tensor->get_partial_shape().rank());
 
-        std::vector<int64_t> split_lengths;
         switch (split_lengths_tensor->get_element_type())
```
Comment (mbencer) [inline ngraph/core/src/op/variadic_split.cpp:193]: The same for `split_lengths_tensor`

**Review by mbencer — CHANGES_REQUESTED** [review]

**Comment by mbencer** [comment]
Probably you can remove reviewers which were added after some unnecessary commits

File: ngraph/core/src/op/split.cpp (hunk 2 above)
Comment (jdanieck) [reply ngraph/core/src/op/split.cpp:225]: I see that split has evaluate() so binary size concern is valid. Why remove it if it was done on purpose?

File: ngraph/core/src/op/variadic_split.cpp (hunk 3 above)
Comment (jdanieck) [reply ngraph/core/src/op/variadic_split.cpp:174]: I see that variadic_split has evaluate() so binary size concern is valid. Why remove it if it was done on purpose?

File: ngraph/core/src/op/variadic_split.cpp (hunk 4 above)
Comment (jdanieck) [reply ngraph/core/src/op/variadic_split.cpp:193]: I see that variadic_split has evaluate() so binary size concern is valid. Why we want to add more types if it was done on purpose?

File: ngraph/core/src/op/split.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/core/src/op/split.cpp:225]: But it doesn't affect binary size (or I miss sth) so why not ;)
I've tested such case for Split op (Ubuntu, Release, libngraph.so)
1. before using `host_tensor_2_vector`: 7772296
2. Using switch to to accept only i32, i64, u64 as in current version: 7768200
3. My suggestion to remove switch and add assert to accept each integral_number: also 7772296
but correct me if I'm wrong

Comment (jdanieck) [reply ngraph/core/src/op/split.cpp:225]: OK, if we don't increase binary size then that's definitely good advise. Let's go with is_integral_number().

**Review by mbencer — APPROVED** [review]

**Review by postrational — APPROVED** [review]

**Pull request merged**